### PR TITLE
PUB-2022 Add pre-authorize check to delete subscription by ID and bulk delete subscription endpoints

### PR DESCRIPTION
**Pull request by KianKwa**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-2022

### Change description ###

Add pre-authorize check to delete subscription by ID and bulk delete subscription endpoints

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionController.java [hunk 1]
```diff
@@ -105,6 +112,23 @@ public ResponseEntity<String> bulkDeleteSubscriptions(@RequestBody List<UUID> su
                                                    .replace("]", "")));
     }
 
+    @ApiResponse(responseCode = OK_CODE, description = "Subscriptions with IDs {subIds} deleted")
+    @ApiResponse(responseCode = AUTH_ERROR_CODE, description = NOT_AUTHORIZED_MESSAGE)
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionController.java:116]: Duplicate response code with line 117

File: src/test/java/uk/gov/hmcts/reform/pip/subscription/management/service/AuthorisationServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+@ExtendWith(MockitoExtension.class)
+class AuthorisationServiceTest {
+    private static final String USER_ID = UUID.randomUUID().toString();
+    private static final String ANOTHER_USER_ID = UUID.randomUUID().toString();
+    private static final UUID SUBSCRIPTION_ID = UUID.randomUUID();
+    private static final UUID SUBSCRIPTION_ID2 = UUID.randomUUID();
+    private static final UUID SUBSCRIPTION_ID3 = UUID.randomUUID();
+
+    private static final String CAN_DELETE_SUBSCRIPTION_MESSAGE = "USer should be able to delete subscription";
```
Comment (ChrisS1512) [inline src/test/java/uk/gov/hmcts/reform/pip/subscription/management/service/AuthorisationServiceTest.java:32]: Upper case S in user :)

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java [hunk 3]
```diff
@@ -11,7 +11,6 @@
 import org.springframework.beans.factory.annotation.Value;
```
Comment (ChrisS1512) [inline src/functionalTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java:11]: I think we should add some tests for the System Admin path too in here. What do you think?

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionController.java (hunk 1 above)
Comment (KianKwa) [reply src/main/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionController.java:116]: Removed the duplicated line

File: src/test/java/uk/gov/hmcts/reform/pip/subscription/management/service/AuthorisationServiceTest.java (hunk 2 above)
Comment (KianKwa) [reply src/test/java/uk/gov/hmcts/reform/pip/subscription/management/service/AuthorisationServiceTest.java]: Corrected typo

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java (hunk 3 above)
Comment (KianKwa) [reply src/functionalTest/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionControllerTests.java:11]: Added delete subscription tests using system admin

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**